scarb/src/bin/scarb/args.rs

```diff
@@ -292,6 +304,46 @@ pub struct GitRefGroup {
     pub rev: Option<String>,
 }
 
+/// Profile specifier.
+#[derive(Parser, Clone, Debug)]
+pub struct ProfileSpec {
+    /// Specify profile to use by name.
+    #[arg(short = 'P', long, value_name = "SPEC")]
+    pub profile: Option<SmolStr>,
+    /// Use release profile.
+    #[arg(long, hide_short_help = true)]
+    pub release: bool,
+    /// Use dev profile.
+    #[arg(long, hide_short_help = true)]
+    pub dev: bool,
+}
+
+impl ProfileSpec {
+    fn verify(&self) -> Result<()> {
+        if self.dev && self.release {
+            bail!("cannot specify both `--dev` and `--release`");
+        }
+        if self.release && self.profile.is_some() {
+            bail!("cannot specify both `--release` and `--profile`");
+        }
+        if self.release && self.dev {
+            bail!("cannot specify both `--release` and `--dev`");
+        }
+        Ok(())
+    }
+
+    pub fn fetch<'a>(&self, ws: &'a Workspace<'_>) -> Result<&'a Profile> {
+        self.verify()?;
+        let profile = if self.release {
+            "release".into()
+        } else {
+            self.profile.clone().unwrap_or("dev".into())
+        };
+
+        ws.current_package()?.fetch_profile(&profile)
+    }
+}
```

Comment:
Looks like this function tries to do too much. Shouldn't this just return the profile name?

```suggestion
    pub fn determine(&self) -> SmolStr {
        match &self {
            Self { release: true, .. } => "release".into(),
            Self { dev: true, .. } => "dev".into(),
            Self { profile: Some(profile), .. } => profile.clone(),
            _ => "dev".into()
        }
    }
}
```